src/builder.py

```diff
@@ -276,8 +279,26 @@ def get_new_executable(self):
         return path
 
 
-class HostBuilder:
-    def __init__(self, build_directory, install_directory):
+class Abstract(NotImplementedError):
+    pass  # TODO: use abc?
+
+
+class Builder:
```

Comment:
Would it make sense to use https://docs.python.org/3/library/abc.html here?

Reply:
yes... hence the TODO.

1st time I tried abc with mypy, mypy didn't grok. But that was a while ago. Let's see if it works now...